### PR TITLE
Add docker-compose for kiwix-server

### PR DESCRIPTION
**Pull request by fazlerabbi37**
Add docker-compose for kiwix-server

- run docker-compose with local file.
- run docker-compose with remote file.

### CODE REVIEW

**Comment by kelson42** [comment]
@fazlerabbi37 Which problem does it solve? Which value does in bring in comparison to a simple `docker run`?

**Comment by fazlerabbi37** [comment]
> @fazlerabbi37 Which problem does it solve? Which value does in bring in comparison to a simple `docker run`?

while it's true that docker-compose is used for managing multiple containers, but it can be useful for single container too. for example:

- quick and easier to define and manage services, networks, volumes etc
- get an app running in one command by just running `docker-compose up`
- fast and simple
- run the app and its related resources isolated network and environment

but for me, it's just easy to manage. I can `cd` into the `kiwix-server` and do `docker-compose up -d`.

**Comment by kelson42** [comment]
@rgaudin I don't know what you think about this, but the value seems really too limited to me to justify we complexify the `README.md`. If you think differently, can you please handle the review process?

**Comment by fazlerabbi37** [comment]
> but the value seems really too limited to me to justify we complexify the README.md

**Comment by rgaudin** [comment]
> @rgaudin I don't know what you think about this, but the value seems really too limited to me to justify we complexify the `README.md`. If you think differently, can you please handle the review process?

To me, the value of this is in documenting the various features of the image. I'd suggest we either keep the docker-compose.yml in the repo as an example or we keep its content into the README.

Or it could even achieve same result by just adding a couple descriptive lines in the *Usage* section of the README…

**Comment by fazlerabbi37** [comment]
@rgaudin can you please check now?

**Review by rgaudin — CHANGES_REQUESTED** [review]
Thanks ; the value of your compose file was in the comments IMO; because we don't document properly this download feature for instance.

So please restore the comment and make sure to end the file with a blank line.

Once this is done, you can rebase it all into one commit

**Comment by fazlerabbi37** [comment]
@rgaudin just to be sure, [docker-compose.yml](https://github.com/fazlerabbi37/kiwix-tools/blob/50290adf321623b1336b68bd006b4a61adc3e687/docker/server/docker-compose.yml) goes back to this state with an empty line at the end? then rebase, right?

**Comment by rgaudin** [comment]
> @rgaudin just to be sure, [docker-compose.yml](https://github.com/fazlerabbi37/kiwix-tools/blob/50290adf321623b1336b68bd006b4a61adc3e687/docker/server/docker-compose.yml) goes back to this state with an empty line at the end? then rebase, right?

Yes

**Review by rgaudin — APPROVED** [review]
Great! Can you rebase from master instead of merging master in the branch instead so we have a single commit ?

**Comment by Teraskull** [comment]
I would suggest adding this example to commands:
```yaml
command:
  - "*.zim"
```
Might be useful if there is a lot of ZIMs, instead of adding each filename manually.

**Comment by rgaudin** [comment]
> I would suggest adding this example to commands:
> 
> ```yaml
> command:
>   - "*.zim"
> ```
> 
> Might be useful if there is a lot of ZIMs, instead of adding each filename manually.

Please do

File: docker/server/docker-compose.yml.example [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      image: kiwix/kiwix-serve
+      # uncomment next 5 lines to use it with local zim file in /tmp/zim
+      # volumes:
+      #   - /tmp/zim:/data
+      # command:
+      #   - wikipedia.zim
+      #   - wiktionary.zim
+      # uncomment next 2 lines to use it with remote zim file
+      # environment:
+      #   - 'DOWNLOAD=https://download.kiwix.org/zim/wikipedia_bm_all.zim'
```
Comment (fazlerabbi37) [inline docker/server/docker-compose.yml.example:15]: @rgaudin @Teraskull is it possible to pass multiple URL via DOWNLOAD variable? if not, how does one use multiple zim file while using remote download function?

Comment (kelson42) [reply docker/server/docker-compose.yml.example:14]: No, it is limited to one ZIM file.

Comment (kelson42) [reply docker/server/docker-compose.yml.example:14]: Have a look to the pretty simple `start.sh` file to understand how it works https://github.com/kiwix/kiwix-tools/blob/master/docker/server/start.sh. You can still specify whatever you want as argument (so a custom `library.xml` for example.

**Comment by fazlerabbi37** [comment]
> Please do

@rgaudin done!

**Comment by Teraskull** [comment]
@fazlerabbi37 Can you also provide a [docker compose](https://docs.docker.com/compose/) link like this?

Since `docker-compose` is v1 and deprecated, and v2 is used as a docker plugin.

**Comment by fazlerabbi37** [comment]
> @fazlerabbi37 Can you also provide a [docker compose](https://docs.docker.com/compose/) link like this?
> 
> Since `docker-compose` is v1 and deprecated, and v2 is used as a docker plugin.

you mean in the README.md?

**Comment by Teraskull** [comment]
@fazlerabbi37 Yes

**Comment by fazlerabbi37** [comment]
> @fazlerabbi37 Yes

done! check now. :smile:

**Pull request merged**